mlir/lib/Dialect/Rock/Transforms/GemmToGridwise.cpp

```diff
@@ -149,29 +150,33 @@ GemmRewritePattern::matchAndRewrite(GemmOp op, GemmOpAdaptor adaptor,
   b = padMatrix(b, rw, loc, "gemmK", extraPad.k, "gemmN", extraPad.n);
   c = padMatrix(c, rw, loc, "gemmM", extraPad.m, "gemmN", extraPad.n);
 
-  bool isXdlops = bitEnumContainsAll(op.getFeatures(), GemmFeatures::mfma);
-  bool isWmma = bitEnumContainsAll(op.getFeatures(), GemmFeatures::wmma);
-
   IntegerAttr blockSize = op.getDerivedBlockSizeAttr();
-  if (isXdlops && !blockSize)
+  bool isAccel = rock::isAccel(op.getFeatures(), a.getType());
+  if (isAccel && !blockSize)
     return op.emitOpError("block size must be set at lowering");
   IntegerAttr gridSize = op.getGridSizeAttr();
   if (!gridSize)
     return op.emitOpError("grid size must be set at lowering");
-  if (isXdlops) {
-    // Onne the attribute copies are gone, make this a replaceOp
-    rw.create<GridwiseGemmAccelOp>(loc, a, b, c, op.getArchAttr(),
-                                   op.getFeaturesAttr(),
-                                   op.getStoreMethodAttr(), blockSize, gridSize,
-                                   params.cast<XdlopsGemmParamsAttr>());
-    rw.eraseOp(op);
-  } else if (isWmma) {
-    // Onne the attribute copies are gone, make this a replaceOp
-    rw.create<GridwiseGemmAccelOp>(loc, a, b, c, op.getArchAttr(),
-                                   op.getFeaturesAttr(),
-                                   op.getStoreMethodAttr(), blockSize, gridSize,
-                                   params.cast<WmmaGemmParamsAttr>());
-    rw.eraseOp(op);
+
+  if (isAccel) {
+    bool isXdlops = bitEnumContainsAll(op.getFeatures(), GemmFeatures::mfma);
+    bool isWmma = bitEnumContainsAll(op.getFeatures(), GemmFeatures::wmma);
+
+    if (isXdlops) {
+      // Onne the attribute copies are gone, make this a replaceOp
+      rw.create<GridwiseGemmAccelOp>(
+          loc, a, b, c, op.getArchAttr(), op.getFeaturesAttr(),
+          op.getStoreMethodAttr(), blockSize, gridSize,
+          params.cast<XdlopsGemmParamsAttr>());
+      rw.eraseOp(op);
+    } else if (isWmma) {
+      // Onne the attribute copies are gone, make this a replaceOp
+      rw.create<GridwiseGemmAccelOp>(
+          loc, a, b, c, op.getArchAttr(), op.getFeaturesAttr(),
+          op.getStoreMethodAttr(), blockSize, gridSize,
+          params.cast<WmmaGemmParamsAttr>());
+      rw.eraseOp(op);
```

Comment:
Can we pass in the parent tuning parameters type but dynamically cast them when in gridwise gemm op? It looks inconvenient that we have to do such kind of branching considering that you already have both ops unified under the same type.

Reply:
Yeah, it shouldn't be needed to have this if statement here. You could go with `isXdlops || isWmma` and cast to the interface